cloud-aws/src/main/resources/templates/aws-cf-stack.ftl

```diff
@@ -495,8 +495,51 @@
     </#if>
     <#if group_has_next>,</#if>
     </#list>
+
+    <#if enableEfs>
+    ,
+    "ElasticFileSystem":{
+      "Type":"AWS::EFS::FileSystem",
+      "Properties":{
+        <#if efsFileSystem.backupPolicyStatus??>
+          "BackupPolicy" : {
+            "Status" : "${efsFileSystem.backupPolicyStatus}"
+          },
+        </#if>
+        "Encrypted" : "${efsFileSystem.encrypted? string("true", "false")}",
+        <#if efsFileSystem.fileSystemPolicy??>
+          "FileSystemPolicy" : ${efsFileSystem.fileSystemPolicy},
+        </#if>
+        <#if efsFileSystem.fileSystemTags??>
+          "FileSystemTags": [
+            <#list efsFileSystem.fileSystemTags?keys as key>
+            {
+              "Key" : "${key}",
+              "Value" : "${efsFileSystem.fileSystemTags[key]}"
+            }<#if (key_index + 1) != efsFileSystem.fileSystemTags?keys?size> ,</#if>
+          </#list>
+          ],
+        </#if>
+        <#if efsFileSystem.kmsKeyId??>
+         "KmsKeyId": "${efsFileSystem.kmsKeyId}",
+        </#if>
+        <#if efsFileSystem.lifeCyclePolicies??>
+          "LifecyclePolicies": [
+            <#list efsFileSystem.lifeCyclePolicies as lifeCyclePolicy>
+              ${lifeCyclePolicy}<#if (lifeCyclePolicy_index + 1) != efsFileSystem.lifeCyclePolicies?size> ,</#if>
+            </#list>
+            ],
+        </#if>
+        "PerformanceMode": "${efsFileSystem.performanceMode}",
+        <#if efsFileSystem.throughputMode == "provisioned">
+          "ProvisionedThroughputInMibps": "${efsFileSystem.provisionedThroughputInMibps}",
+        </#if>
+        "ThroughputMode": "${efsFileSystem.throughputMode}"
+      }
+    }
+    </#if>
   }
-  <#if mapPublicIpOnLaunch || (enableInstanceProfile && !existingRole)>
+  <#if mapPublicIpOnLaunch || (enableInstanceProfile && !existingRole) || !existingVPC || !existingSubnet || enableEfs>
```

Comment:
What's the rationale for adding the `!existingVPC || !existingSubnet` checks here?

Based on my talks with Krisztian `existingVPC` and `existingSubnet` are always true, because network creation happens in an earlier part of the code. So by the time the code gets here there is already an existing network, and those will always evaluate to false anyway.

Reply:
In some testing cases, it may not be true.